### PR TITLE
Implement ZLIB in pure Kotlin

### PR DESCRIPTION
**Pull request by nodh**
Implemented by gpt-5.3-codex high

### CODE REVIEW

File: vck/src/commonTest/kotlin/at/asitplus/wallet/lib/DefaultZlibServiceOkioInteropTest.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import okio.Buffer
+import okio.buffer
+import okio.deflate
+import okio.inflate
+import kotlin.random.Random
+
+val DefaultZlibServiceOkioInteropTest by testSuite {
+    "ours compressed data inflates with okio" {
```
Comment (JesusMcCloud) [inline vck/src/commonTest/kotlin/at/asitplus/wallet/lib/DefaultZlibServiceOkioInteropTest.kt]: i'd wrap all these tests inside a `forAll` with bytearray arbs. this would also cover edge cases (while allowing to exclude zero) by default. otherwise it looks sane and if the tests check out it should be fine

**Review by JesusMcCloud — CHANGES_REQUESTED** [review]

**Review by JesusMcCloud — APPROVED** [review]
